### PR TITLE
[Notifications] feat: add functionality for the users to toggle their global notifications

### LINKED ISSUE #3003: Notifications: Allow users to toggle the notifications feature on/off
### Story

As a *user*, I want to *be able to turn notifications on and off*, so that *I can control whether or not I receive notifications*

### Description

The user should be able to turn the notifications feature on and off for themselves in their user preferences. Doing so should update the notifications data on the user object in our db to store this preference, but should not delete the Magicbell user. This ticket should cover all device widths.

### Design

[Figma link](https://www.figma.com/design/0Fi3AaDlr3LAXBfsOXBGyk/User-Account?node-id=6592-9447&t=RpPiEaurGEYf4K5z-4)

Image

### This issue is complete when...

- [ ] Users can update their preference to turn off notifications app wide, and this preference is stored in our db.

### PR DESCRIPTION
**Pull request by bassgeta**
## Description

This PR introduces a new section to the user's profile `Advanced` section. Currently it only added data about handling notifications.
[Figma link](https://www.figma.com/design/0Fi3AaDlr3LAXBfsOXBGyk/User-Account?node-id=6592-9447&t=RpPiEaurGEYf4K5z-4)
image

## Testing
This PR basically just refactors smelly code and implements a simple switch that enables notifications.
Do note that the API stores data about disabled state, that's why there's some `!` action in the `NotificationsToggle` component.

If you haven't tested any notifications related PRs do the following first:
Add
```
MAGICBELL_API_KEY=ask me or @davecreaser for both of these 
MAGICBELL_API_SECRET=
```
to `.env.local.secrets` and run `npm run prepare`

### Notifications settings 
1. As `leela` navigate to `http://localhost:9091/account/advanced` and verify that the notifications default to enabled
image
2. Upon clicking the switcher, it should change the flag to false and display a toast
image
3. Now go ahead and create a `Simple payment` action sending some CREDS to `leela`
image
4. Verify that you don't get an unread notification counter in the user hub
image
5. Next, go back to `http://localhost:9091/account/advanced` and toggle the notifications on, a toast should be displayed
image
6. Repeat the simple payment action (notice the little blue dot there now :eyes: ) 
image
7. Verify that you have unread notifications 
image

### Metatransactions
Just verify that they still work :D 
1. Login into `metamask` as `leela`'s dev wallet and toggle metatransactions, the toast should show up and the changes should persist when refreshing
image
2. Create any action and verify that you only need to sign the transactions instead of paying the gas fee
image
3. Go back and toggle metatransactions off, the toast should show up and the changes still persist
image
4. Create an action again, notice that you get prompted for paying gas fees
image

## Diffs

**New stuff** ✨

* `SettingsRow` component which handles all the sections, titles, tooltips, borders, etc
* `NotificationsSection` which has the toggle for notifications enabled
* `UpdateUserGlobalNotificationsMuted` mutation

**Changes** 🏗

* Reworked `FeesForm` to use the new `SettingsRow` component and refactor it 

**Deletions** ⚰️

* Remove `RpcForm`
* Remove old `SettingsRow`

Resolves  #3003

### CODE REVIEW

**Review by rdig — CHANGES_REQUESTED** [review]
Functionality wise, and even code wise this is all good, however `RpcForm` itself is still needed as it's part of the whole "D" in the CDapp which stands for Decentralized.

It's basically there to allow you to run this locally, so you're able to provide a RPC endpoint that you host yourself, and have the app run on that.

Granted, this was never properly wired up in the settings, but at some point we will _(when the focus will be less on shipping massing amount of features)_

So even if you remove it from the settings page, the `RpcForm` still needs to exist in our codebase as a component.

That's the only thing for which I rejected this, other than that, this is all good. 

Notifications disabled:


https://github.com/user-attachments/assets/f42e6054-3496-4611-b462-1362d7dc1c94

Notifications enabled:

https://github.com/user-attachments/assets/5afe310f-0cc4-4872-9455-09fe49327771

Notifications and Metatransactions Enabled:

https://github.com/user-attachments/assets/deeaac24-07bc-48d6-85fa-9edab56aaaca

**Review by rdig — APPROVED** [review]
'RpcForm' is back in, so I'm happy.

Thank you for reverting that! :100:

**Review by iamsamgibbs — APPROVED** [review]
This is working very nicely, great job @bassgeta The SettingsRow components are very nice.

Notifications are enabled by default:

Screenshot 2024-09-25 at 09 06 20

Disabling them shows the toast:

Screenshot 2024-09-25 at 09 06 58

Making a new transaction does not increase the number of notifications (so no new ones were created)

Screenshot 2024-09-25 at 09 09 55

And the correct toast is shown when reenabling:

Screenshot 2024-09-25 at 09 10 08

Then the notification count does increase:

Screenshot 2024-09-25 at 09 10 38

Also tested the metatransactions are still functioning, which they are 👍 

Screenshot 2024-09-25 at 09 11 38

Screenshot 2024-09-25 at 09 13 03

The only issue I have is that I believe the notifications tab should disappear entirely from the UserHub when notifications are disabled (correct me if I'm wrong @davecreaser ). The blue dot should also not show.

It should just require updating the NotificationsContext with this preference from the database. The UserHub tab already has a hardcoded `notificationsServiceIsEnabled` variable that would just need the value from the NotificationsContext, I don't recall if I setup the same for the blue dot or not, but would be a relatively simple change.

All that said I think it may be better done as it's own issue unless you fancy adding it here.

File: src/components/v5/common/SettingsRow/index.ts [hunk 1]
```diff
@@ -1 +1,17 @@
...
+const SettingsRow = {
+  Container: SettingsRowContainer,
+  Content: SettingsRowContent,
+  Description: SettingsRowDescription,
+  Subtitle: SettingsRowSubtitle,
+  Title: SettingsRowTitle,
+  Tooltip: SettingsRowTooltip,
+};
+
+export default SettingsRow;
```
Comment (davecreaser) [inline src/components/v5/common/SettingsRow/index.ts:17]: Very nice, very clean way of handling these!

**Comment by davecreaser** [comment]
Nice one @bassgeta! Works great, and the code is once again much cleaner 😆 You're really on a mission to clean up the codebase haha, every PR seems to improve something 🚀 

Tested the notifications toggle and all is good:
Screenshot 2024-09-25 at 14 46 45
Screenshot 2024-09-25 at 14 47 20
Screenshot 2024-09-25 at 14 51 34
Screenshot 2024-09-25 at 14 54 09
Screenshot 2024-09-25 at 14 55 11

Metatransactions also all working, but I forgot to take screenshots 🙃 

About the comment from @iamsamgibbs about it not showing the tab, he's correct (and it also shouldn't connect to Magicbell at all in the context provider in that case). I'll make a new ticket to handle that though so we can merge this, since I didn't put it in the original ticket 👍

**Review by davecreaser — APPROVED** [review]

**Pull request merged**